docs/languages/csharp/event-pattern.md

```diff
@@ -1,9 +1,344 @@
 # The Standard .NET Event Pattern
 
+By [Bill Wagner](https://github.com/BillWagner)
+
 [Previous](events-overview.md)
 
-This article will the standard .NET event pattern.
+.NET events generally follow a few known patterns. Standardizing
+on these patterns means that developers can leverage knowledge of
+those standard patterns can be applied to any .NET event program.
+
+Let's go through these standard patterns so you will have all
+the knowledge you need to create standard event sources, and
+subscribe and process standard events in your code.
+
+# Event Delegate Signatures
+
+The standard signature for a .NET event delegate is:
+
+```cs
+void Handler(object sender, EventArgs args);
+```
+
+The return type is void. Events are based on delegates and are
+multicast delegates. That supports multiple subscribers for any
+event source. The single return value from a method doesn't scale
+to multiple event subscribers. Which return value does the event
+source see after raising an event? Later in this article you'll
+see how to create event protocols that support event subscribers
+reporting information to the event source.
+
+The argument list contains two arguments: the sender, and the event
+arguments. The compile time type of `sender` is `System.Object`,
+even though you likely know a more derived type that would always
+be correct. By convention, use `object`.
+
+The second argument has typically been a type that is derived from
+`System.EventArgs`. (You'll see in the 
+[next section](modern-events.md) that this convention is no longer
+enforced.) If your event type does not need any additional
+arguments, you will still provide both arguments.
+There is a special value, `EventArgs.Empty` that you should use to
+denote that your event does not contain any additional information.
+
+Let's build a class that lists files in a directory, or any
+subdirectory that follow a pattern. This component raises an event
```

Comment:
suggested revision: files in a directory or any of its subdirectories that follow a pattern